tests/stress/stress_test.go

```diff
@@ -641,3 +642,44 @@ func TestSync_SubmitExtrinsicLoad(t *testing.T) {
 		// Output: Balance transferred from Alice to Bob: 100000000000000
 	}
 }
+
+func TestStress_SecondarySlotProduction(t *testing.T) {
+	numNodes := 3
+
+	nodes, err := utils.InitializeAndStartNodes(t, numNodes, utils.GenesisThreeAuths, utils.ConfigDefault)
+	require.NoError(t, err)
+
+	primaryCount := 0
+	secondaryCount := 0
+	otherCount := 0
+	for i := 1; i < 20; i++ {
+		fmt.Printf("%d iteration\n", i)
+		hash, err := utils.GetBlockHash(t, nodes[0], fmt.Sprintf("%d", i))
+		require.NoError(t, err)
+
+		block := utils.GetBlock(t, nodes[0], hash)
+		header := block.Header
+
+		preDigestItem := header.Digest.Types[0]
+
+		preDigest, ok := preDigestItem.Value().(gosstypes.PreRuntimeDigest)
+		require.True(t, ok)
+
+		babePreDigest, err := gosstypes.DecodeBabePreDigest(preDigest.Data)
+		require.NoError(t, err)
+
+		switch babePreDigest.(type) {
+		case gosstypes.BabePrimaryPreDigest:
+			primaryCount++
+		case gosstypes.BabeSecondaryVRFPreDigest:
+			otherCount++
+		case gosstypes.BabeSecondaryPlainPreDigest:
+			secondaryCount++
+		}
+		require.NotEqual(t, babePreDigest, nil)
+
+		time.Sleep(10 * time.Second)
```

Comment:
Do we need to sleep that long here 🤔

Reply:
yes, I have tried with 5 sec, which does not work. May be there is a perfect number, but it is better to keep a few seconds extra to ensure few failed attempts.